### DISCUSSION TITLE
Using blueprintjs with Vanilla JS, Radio group not working, is this a bug?

### DISCUSSION THREAD

**Discussion by almosnow** [palantir/blueprint#5748] [Q&A] [answered]
I am using blueprintjs with Vanilla JS, as described here: https://blueprintjs.com/docs/#blueprint/getting-started.vanilla-js-apis

Then I create a RadioGroup with Radio buttons like this:

```
var e = React.createElement(Blueprint.Core.RadioGroup, {
	label:'Meal Choice',
	onChange:function(e) { console.log('change', e.currentTarget.value); }
}, [
	React.createElement(Blueprint.Core.Radio, {
		label:'Soup',
		value:1
	}),
	React.createElement(Blueprint.Core.Radio, {
		label:'Salad',
		value:2
	}),
	React.createElement(Blueprint.Core.Radio, {
		label:'Sandwich',
		value:3
	})
]);

ReactDOM.render(e, document.body.appendChild(document.createElement('div')));
```
And get a component that looks like this:

![image](https://user-images.githubusercontent.com/4924797/137529158-c87bdbc6-8288-41c5-926c-1472bcf32f69.png)

Now, whenever I click any of the radio buttons, it just stays like that, i.e. it doesn't show/render the current option as selected. The callback for the onChange event gets called with the correct value, so that's weird.

Is this a bug or am I missing something in the implementation?

**Comment by adidahiya** [comment, marked as answer]
It looks like you never update any of the radio components to be marked as selected in the `onChange` event handler. You need to set `RadioGroup`'s `selectedValue` prop.

Please try to produce a bug repro in the code sandbox link provided in the issue template, if possible. If no such repro is possible, then it's likely not a bug.

**Comment by almosnow** [comment]
I saw the sandbox but it ws not set up to work w/ Vanilla JS as in the provided example, I could still share a codepen or something if you think that's convenient.

>It looks like you never update any of the radio components to be marked as selected in the onChange event handler. You need to set RadioGroup's selectedValue prop.

That's what I've been trying to do w/ no avail, could you provide an example of this? Sorry if i'm being kind of a noob here but well ... that what's I am. Also, because of the particular context where we are using this, we have to resort to Vanilla JS, hence why I'm trying to get this to work.

**Comment by almosnow** [comment]
???
